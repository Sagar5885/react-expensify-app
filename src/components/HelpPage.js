import React from 'react';

const HelpPage = () => (
    <div>
        This is from my Help Expanse component 
    </div>
);

export default HelpPage;