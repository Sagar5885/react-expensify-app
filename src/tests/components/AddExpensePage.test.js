import React from 'react';
import { shallow } from 'enzyme';
import { AddExpansePage } from '../../components/AddExpansePage';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddExpansePage onSubmit={onSubmit} history={history} />);
});

test('should render AddExpansePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
});