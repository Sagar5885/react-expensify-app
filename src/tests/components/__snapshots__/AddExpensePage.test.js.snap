// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensePage correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Add Expense
    </h1>
    <ExpenseForm
        onSubmit={[Function]}
    />
</div>,
  "nodes": Array [
    <div>
      <h1>
            Add Expense
      </h1>
      <ExpenseForm
            onSubmit={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <AddExpensePage
      addExpense={[Function]}
      history={
            Object {
                  "push": [Function],
                }
      }
/>,
    "_instance": AddExpensePage {
      "context": Object {},
      "onSubmit": [Function],
      "props": Object {
        "addExpense": [Function],
        "history": Object {
          "push": [Function],
        },
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <h1>
            Add Expense
      </h1>
      <ExpenseForm
            onSubmit={[Function]}
      />
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AddExpensePage
    addExpense={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
}
`;
